IllegalStateException when bulk deleting notes

```
java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:300)
            at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
            at java.util.concurrent.FutureTask.run(FutureTask.java:242)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.lang.IllegalStateException: Couldn't read row 0, col 0 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
            at android.database.CursorWindow.nativeGetLong(Native Method)
            at android.database.CursorWindow.getLong(CursorWindow.java:511)
            at android.database.AbstractWindowedCursor.getLong(AbstractWindowedCursor.java:75)
            at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:220)
            at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:162)
            at android.database.CursorWrapper.moveToPosition(CursorWrapper.java:162)
            at com.automattic.simplenote.NoteListFragment$NotesCursorAdapter.getItem(NoteListFragment.java:508)
            at com.automattic.simplenote.NoteListFragment$trashNotesTask.doInBackground(NoteListFragment.java:714)
            at com.automattic.simplenote.NoteListFragment$trashNotesTask.doInBackground(NoteListFragment.java:705)
            at android.os.AsyncTask$2.call(AsyncTask.java:288)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
```

--- a/Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java
+++ b/Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java
@@ -709,14 +709,22 @@ public class NoteListFragment extends ListFragment implements AdapterView.OnItem
         @Override
         protected Void doInBackground(Void... args) {
             SparseBooleanArray selectedRows = getListView().getCheckedItemPositions();
+
+            // Get the checked notes and add them to the deletedNotesList
+            // We can't modify the note in this loop because the adapter could change
+            List<Note> deletedNotesList = new ArrayList<Note>();
             for (int i = 0; i < selectedRows.size(); i++) {
-                if (selectedRows.valueAt(i) == true) {
-                    Note deletedNote = mNotesAdapter.getItem(selectedRows.keyAt(i));
-                    deletedNotesIds.add(deletedNote.getSimperiumKey());
-                    deletedNote.setDeleted(!deletedNote.isDeleted());
-                    deletedNote.setModificationDate(Calendar.getInstance());
-                    deletedNote.save();
+                if (selectedRows.valueAt(i)) {
+                    deletedNotesList.add(mNotesAdapter.getItem(selectedRows.keyAt(i)));
                 }
+            }
+
+            // Now loop through the notes list and mark them as deleted
+            for (Note deletedNote : deletedNotesList) {
+                deletedNotesIds.add(deletedNote.getSimperiumKey());
+                deletedNote.setDeleted(!deletedNote.isDeleted());
+                deletedNote.setModificationDate(Calendar.getInstance());
+                deletedNote.save();
             }
 
             return null;

